File: src/my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using std::placeholders::_1;
using std::placeholders::_2;

using std::bind;

using example_interfaces::srv::AddTwoInts;

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_oop");
    auto client = node->create_client<AddTwoInts>("add_two_ints");
    while(!client->wait_for_service(std::chrono::seconds(1)))
        RCLCPP_WARN(node->get_logger(), "Waiting for Two Ints server to be up");

    auto request = std::make_shared<AddTwoInts::Request>();
    request->a = 3;
    request->b = 8;

    auto future = client->async_send_request(request);
    if(rclcpp::spin_until_future_complete(node, future) == rclcpp::executor::FutureReturnCode::SUCCESS)
        RCLCPP_INFO(node->get_logger(), "Response is %d for sum %d + %d", future.get()->sum, request->a, request->b);
    else
        RCLCPP_ERROR(node->get_logger(), "Error while calling server for sum %d + %d", request->a, request->b);

    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts

def main(args=None):
    rclpy.init(args=args) # first line to be written in any ROS2 .py node
    node = Node("add_two_ints_no_oop")
    client = node.create_client(AddTwoInts, "add_two_ints")
    
    while(not client.wait_for_service(1.0)):
        node.get_logger().warn("Waiting for server add_two_ints to be up")
    
    '''Service is up now'''
    request = AddTwoInts.Request()
    request.a = 3
    request.b = 7

    future = client.call_async(request)
    rclpy.spin_until_future_complete(node, future)
    
    try:
        response = future.result()
        node.get_logger().info("Response is {0} for request \"{1} + {2}\"".format(response.sum, request.a, request.b))

    except Exception as e:
        node.get_logger().error("Service call failed %r" %(e,))


    rclpy.shutdown() # last line of any ROS2 .py node

if __name__ == "__main__":
    main()